src/Libraries/Microsoft.Extensions.Telemetry/README.md

```diff
@@ -18,6 +18,90 @@ Or directly in the C# project file:
 </ItemGroup>
 ```
 
+## Usage
+
+### Service Log Enrichment
+
+Enriches logs with application-specific information based on `ApplicationMetadata` information. The bellow calls will add the service log enricher to the service collection.
+
+```csharp
+// Add service log enricher with default settings
+builder.Services.AddServiceLogEnricher();
+
+// Or configure with options
+builder.Services.AddServiceLogEnricher(options =>
+{
+    options.ApplicationName = true;
+    options.BuildVersion = true;
+    options.DeploymentRing = true;
+    options.EnvironmentName = true;
+});
+```
+
+### Latency Monitoring
+
+Provides tools for latency data collection and export. The bellow example uses the built-in Console exporter, but custom exporters can be created by implementing the `ILatencyDataExporter` interface.
+
+```csharp
+// Add latency console data exporter with configuration
+builder.Services.AddConsoleLatencyDataExporter(options =>
+{
+    options.OutputCheckpoints = true;
+    options.OutputMeasures = true;
+    options.OutputTags = true;
+});
+```
+
+In order for the latency data to be exported, a call to `ILatencyDataExporter.ExportAsync()` is required. This can either be called manually, or by using the Request Latency Middleware inside the `Microsoft.AspNetCore.Diagnostics.Middleware` package by adding:
+
+```csharp
+// Add Latency Context
+builder.Services.AddLatencyContext();
+
+// Add Checkpoints, Measures, Tags
+builder.Services.RegisterCheckpointNames("databaseQuery", "externalApiCall");
+builder.Services.RegisterMeasureNames("responseTime", "processingTime");
+builder.Services.RegisterTagNames("userId", "transactionId");
+
+// Add Console Latency exporter.
+builder.Services.AddConsoleLatencyDataExporter();
+// Optionally add custom exporters.
+builder.Services.AddSingleton<ILatencyDataExporter, MyCustomExporter>();
+
+// Add Request latency telemetry.
+builder.Services.AddRequestLatencyTelemetry();
+
+// ...
+
+// Add Request Latency Middleware which will automatically call ExportAsync on all registered latency exporters.
+app.UseRequestLatencyTelemetry();
+```
+
+### Logging Enhancements
+
+Offers additional logging capabilities like stack trace capturing, exception message inclusion, and log redaction.
+
+```csharp
+// Enable log enrichment.
+builder.EnableEnrichment(options =>
+{
+    options.CaptureStackTraces = true;
+    options.IncludeExceptionMessage = true;
+    options.MaxStackTraceLength = 500;
+    options.UseFileInfoForStackTraces = true;
+});
+
+builder.Services.AddServiceLogEnricher(); // <- This call is required in order for the enricher to be added into the service collection.
+
+// Enable log redaction
+builder.EnableRedaction(options =>
+{
+    options.ApplyDiscriminator = true;
+});
+
+builder.Services.AddRedaction(); // <- This call is required in order for the redactor provider to be added into the service collection.
```

Comment:
IIRC this call requires `Microsoft.Extensions.Compliance.Redaction` to be referenced